lib/index.js

```diff
@@ -133,13 +133,12 @@ api.validate = function(name, data, callback) {
     return api.validateInstance(data, schemas.body, callback);
   }
 
-  // schema does not exist, return middle that raises error
+  // schema does not exist
   if(notFound) {
-    return (req, res, next) => {
-      next(new BedrockError(
-        'Could not validate request; unknown schema name (' + notFound + ').',
-        'UnknownSchema', {schema: notFound}));
-    };
+    const err = new BedrockError(
+      'Could not validate request; unknown schema name (' + notFound + ').',
+      'UnknownSchema', {schema: notFound});
+    throw err;
```

Comment:
I think this whole section should be moved up before the `arguments.length > 1` check. If the schema is not found, we should create this error and then -- either throw it if there's no callback or pass it to the callback and return if there is. Then -- everything else should proceed from there. That would consolidate this code block and error to one place instead of two -- and deal with calling validation early.